pkg/registry/core/componentstatus/rest.go

```diff
@@ -38,12 +38,12 @@ import (
 )
 
 type REST struct {
-	GetServersToValidate func() map[string]*Server
+	GetServersToValidate func() (map[string]Server, error)
 	rest.TableConvertor
 }
 
 // NewStorage returns a new REST.
-func NewStorage(serverRetriever func() map[string]*Server) *REST {
+func NewStorage(serverRetriever func() (map[string]Server, error)) *REST {
```

Comment:
returning and error this means that something that always worked before it may start to fail , also to stop apiserver bootstrap, no?

Reply:
This is just repackaging an error that prevents creation etcd client creation vs http client reported lazely. One example of such error that comes to my mind is validating TLS config. Apiserver bootstrap is not affected as this function is called only when componentstatuses are called.